### PR TITLE
fix(NewMessage, EditabeTextField) - don't parse NcRichContenteditable output before sending

### LINKED ISSUE #10227: Pasted URL in Talk gets changed once posted
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Open a Talk room
2. Paste the following link:
  ```
  https://instance.com/xwiki/bin/admin/XWiki/XWikiPreferences?editor=globaladmin&section=Invitation
  ```
4. Hit enter

### Expected behaviour
The exact same link should be displayed and a preview generated

### Actual behaviour
https://.com/xwiki/bin/admin/XWiki/XWikiPreferences?editor=globaladmin§ion=Invitation

The "&sec" is replaced by a weird "§" character

## Talk app

**Talk app version:** (see apps admin page: `/index.php/settings/apps`) c.nc.c

**Custom Signaling server configured:** yes/no and version (see additional admin settings: `/index.php/index.php/settings/admin/talk#signaling_server`)

**Custom TURN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk#turn_server`)

**Custom STUN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk#stun_server`)

## Browser

Firefox Windows latest stable

**Microphone available:** yes/no

**Camera available:** yes/no

**Operating system:** Windows/Ubuntu/...

**Browser name:** Firefox/Chrome/...

**Browser version:** 85/96/...

### Browser log

<details>
```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

## Server configuration

**Operating system**: Ubuntu/RedHat/...

**Web server:** Apache/Nginx

**Database:** MySQL/Maria/SQLite/PostgreSQL

**PHP version:** 8.0/8.1/8.2

**Nextcloud Version:** (see admin page)

**List of activated apps:**

<details>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your server installation folder
```
</details>

**Nextcloud configuration:**

<details>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder
```
</details>

### Server log (data/nextcloud.log)
<details>

```
Insert your server log here
```
</details>

### LINKED ISSUE #10412: Markdown rendering "eats" html tags
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Use talk rc that has markdown support
2. send "\<test\>" as a message

### Expected behaviour
The "\<test\>" text is visible

### Actual behaviour
An empty message shows up

Some properly formatted html messages ("\<b\>test\</b\>") only have the tags stripped, while others  ("\<h1\>test\</h1\>") have both the tag and the content removed during rendering.

Instead of completely removing unsupported tags, it should instead render the html "source" as plain text

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix #10227
* Follow-ups FIXME in the https://github.com/nextcloud-libraries/nextcloud-vue/issues/4492

### 🖼️ Screenshots
Input:
```
&amp &lt &gt &sect
&amp; &lt; &gt; &sect;
https://instance.com/?editor=globaladmin&section=Invitation
```

Output:
🏚️ Before | 🏡 After
---|---
image | image
image | image
``` & < > § & < > § https://instancae.com/?editor=globaladmin§ion=Invitation ``` | ```&amp &lt &gt &sect & < > § https://instancae.com/?editor=globaladmin&section=Invitation ```

### 🚧 Tasks

- [ ] Code review
- [ ] Visual check

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required (should be tested upstream)
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: src/store/messagesStore.js [hunk 1]
```diff
@@ -1145,6 +1145,8 @@ const actions = {
 	 * @param {object} data.options post request options.
 	 */
 	async postNewMessage(context, { temporaryMessage, options }) {
+		context.dispatch('addTemporaryMessage', temporaryMessage)
```
Comment (nickvergessen) [inline src/store/messagesStore.js:1148]: Does it behave correctly for broadcasted messages in breakout rooms?

**Review by DorraJaouad — CHANGES_REQUESTED** [review]
It seems like an issue with text storing in NcRichContenteditable ?
https://github.com/nextcloud/spreed/assets/84044328/4c82418c-6b6d-48f3-8d48-89ce636b915c

**Comment by Antreesy** [comment]
> It seems like an issue with text storing in NcRichContenteditable ?

Yes, that's how the component gives away the output, it escapes all special symbols for security reasons

**Comment by Antreesy** [comment]
P.S. updated PR, add parsing of special symbols to the "Copy formatted message" as example

Either we:
 - do the same for description, leftSidebar, quotes, etc wherever we can get an output of NcRichContenteditable
 - or fix upstream issue https://github.com/nextcloud-libraries/nextcloud-vue/issues/4492, and don't escape special symbols, but that requires more expertise as it could make some security breaches

**Comment by Antreesy** [comment]
P.S. updated PR, raised and referenced an upstream issue

**Review by nickvergessen — APPROVED** [review]
Seems to work

**Pull request merged**